perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -14,50 +15,174 @@ has 'whitespace' => (
     default => sub { Integrations::Whitespace->new(%{shift->feature('whitespace')}) },
 );
 
+sub postcode_database_file {
+    FixMyStreet->path_to('../data/bexley-ww-postcodes.sqlite');
+}
+
 sub bin_addresses_for_postcode {
     my ($self, $postcode) = @_;
 
-    my $addresses = $self->whitespace->GetAddresses($postcode);
+    return [] unless -e postcode_database_file();
+
+    my $db = DBI->connect( 'dbi:SQLite:dbname=' . postcode_database_file(),
+        undef, undef )
+        or return [];
+
+    # Remove whitespaces, make sure uppercase
+    $postcode =~ s/ //g;
+    $postcode = uc $postcode;
+
+    my $addresses = $db->selectall_arrayref(
+        <<SQL,
+   SELECT p.uprn uprn,
+          p.usrn usrn,
+
+          pao_start_number,
+          pao_start_suffix,
+          pao_end_number,
+          pao_end_suffix,
+          pao_text,
+
+          sao_start_number,
+          sao_start_suffix,
+          sao_end_number,
+          sao_end_suffix,
+          sao_text,
+
+          street_descriptor,
+          locality_name,
+          town_name,
+
+          parent_uprn
+     FROM postcodes p
+     JOIN street_descriptors sd
+       ON sd.usrn = p.usrn
+     LEFT OUTER JOIN child_uprns cu
+       ON cu.uprn = p.uprn
+    WHERE p.postcode = ?
+      AND p.uprn NOT IN (
+        SELECT parent_uprn FROM child_uprns
+      )
+SQL
+        { Slice => {} },
+        $postcode,
+    );
 
-    my $data = [ map {
-        {
-            value => $_->{AccountSiteId},
-            label => FixMyStreet::Template::title($_->{SiteShortAddress}) =~ s/^, //r,
-        }
-    } @$addresses ];
+    my @data =
+        map {
+            my $address_string = _build_address_string($_);
+            {
+                value => $_->{uprn},
+                label => FixMyStreet::Template::title( $address_string )
+            };
+        } sort _sort_addresses @$addresses;
```

Comment:
This code looks good, but adds quite a lot of incidental complexity to this file.

I wonder if it would be possible to pull this code (and the related methods below) into a separate `BexleyLLPG` (or whatever) class which encapsulates all the SQLite interactions? Then this `bin_addresses_for_postcode` could simply create a new instance of that class and call a method on it with a postcode and get back the relevant UPRN and address strings.

This is a similar pattern to the `BromleyParks` module, which just exposes a `lookup($search)` method.